plugins/modules/pf.py

```diff
@@ -0,0 +1,303 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2023, Florian Paul Azim Hoberg @gyptazy <gyptazy@gyptazy.ch>
+#
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: pf
+version_added: 2.0.0
+short_description: Manage BSD Packet Filter (pf)
+description:
+    - This module manages the BSD Packet Filter (pf).
+options:
+  config:
+    description:
+      - Path to a pf rule set.
+    type: str
+    required: true
+  filter:
+    description:
+      - If state is C(ignore), filters are ignored.
+      - If state is C(filter), only filter rules will be loaded.
+      - If state is C(nat), only nat rules will be loaded.
+      - If state is C(options), only options rules will be loaded.
+    choices: [ 'ignore', 'filter', 'nat', 'options' ]
+    type: str
+    default: ignore
+  action:
```

Comment:
Modules should be idempotent, and thus not have actions, but states that they ensure.